### PR TITLE
add test for delete_files_in_range

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Test delete_files_in_range with snapshot and iterator.

### CODE REVIEW

File: tests/test_delete_files_in_range.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    // sst2 [3, 6)
+    db.flush(true).unwrap();
+
+    for i in 6..9 {
+        let k = format!("key{}", i);
+        let v = format!("value{}", i);
+        db.put(k.as_bytes(), v.as_bytes()).unwrap();
+        assert_eq!(v.as_bytes(), &*db.get(k.as_bytes()).unwrap().unwrap());
+    }
+    // sst2 [6, 9)
```
Comment (huachaohuang) [inline tests/test_delete_files_in_range.rs]: s/sst2/sst3

File: tests/test_delete_files_in_range.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    // sst3 [6, 9)
+    db.flush(true).unwrap();
+
+    // construct iterator before DeleteFilesInRange
+    let mut iter = db.iter();
+    assert!(iter.seek(SeekKey::Start));
+
+    // delete sst2
+    db.delete_file_in_range(b"key2", b"key7").unwrap();
+    thread::sleep(time::Duration::from_secs(1));
```
Comment (BusyJay) [inline tests/test_delete_files_in_range.rs]: Why sleep?

File: tests/test_delete_files_in_range.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+#[test]
+fn test_delete_files_in_range_with_snap() {
+    let path = TempDir::new("_rust_rocksdb_test_delete_files_in_range_with_snap").expect("");
+    let path_str = path.path().to_str().unwrap();
+    let mut opts = DBOptions::new();
+    opts.create_if_missing(true);
+    let mut cf_opts = ColumnFamilyOptions::new();
+
+    // DeleteFilesInRange ignore sst files in level 0.
+    cf_opts.set_level_zero_file_num_compaction_trigger(1);
```
Comment (BusyJay) [inline tests/test_delete_files_in_range.rs]: Can the initialization be wrapped in a function?

File: tests/test_delete_files_in_range.rs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        let v = format!("value{}", i);
+        db.put(k.as_bytes(), v.as_bytes()).unwrap();
+        assert_eq!(v.as_bytes(), &*db.get(k.as_bytes()).unwrap().unwrap());
+    }
+    // sst3 [6, 9)
+    db.flush(true).unwrap();
+
+    // construct iterator before DeleteFilesInRange
+    let mut iter = db.iter();
+    assert!(iter.seek(SeekKey::Start));
```
Comment (BusyJay) [inline tests/test_delete_files_in_range.rs]: I don't think you should seek before delete files.

File: tests/test_delete_files_in_range.rs (hunk 3 above)
Comment (zhangjinpeng1987) [reply tests/test_delete_files_in_range.rs]: I think in test, this will make it more obvious.

Comment (BusyJay) [reply tests/test_delete_files_in_range.rs]: I don't think so. The two blocks are basically the same, a function name like `initial_data_for_delete_range` is also very verbose.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**